ansible/roles/mcx/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+##############################################################################################
+### playbook to deploy the mcx switch
+### Use this playbook to deploy the console configurations of mcx switch in SONiC testbed
+### This playbook now supports Nokia-7215 and Celestica-E1031-T48S4 only
+### This playbook will configure management IP (default to Ethernet47 for Nokia-7215) and setup 
+### console links based on connection graph.
+################################################################################################
+- name: Gathering lab graph facts about the device
+  conn_graph_facts: host={{ inventory_hostname }}
+  delegate_to: localhost
+  tags: always
+
+- name: Check that OS is SONiC
+  fail: msg="Cannot confirm OS is SONiC"
+  when: os is not defined or os != "sonic"
+
+- name: Set HwSku
+  set_fact:
+    HwSku: "{{ device_info[inventory_hostname][\"HwSku\"] }}"
+
+- name: Check that HwSku is Nokia-7215 or Celestica-E1031-T48S4
+  fail: msg="Device HwSku is not Nokia-7215 or Celestica-E1031-T48S4"
+  when: HwSku != "Nokia-7215" and HwSku != "Celestica-E1031-T48S4"
+
+- name: Prepare mgmt device admin login info
+  set_fact: ansible_ssh_user={{ mgmt_admin_user }} ansible_ssh_pass={{ mgmt_admin_password }}
+
+- name: Move json config to console server
+  template:
+    src: console_config.j2
+    dest: /tmp/base_config.json
+
+- name: Backup config_db.json
+  shell: cp /etc/sonic/config_db.json /etc/sonic/config_db.json.bak
+
+- name: Generate new config_db.json with sonic-cfggen
+  shell: sonic-cfggen -H -k {{ HwSku }} -a "{\"DEVICE_METADATA\":{\"localhost\":{\"hostname\":\"{{ inventory_hostname }}\"}}}" -j /tmp/base_config.json --print-data > /etc/sonic/config_db.json
```

Comment:
I think you can move `DEVICE_METADATA` to the j2 template, and rename the `console_config.j2` to `mcx_config.j2`.